File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/exceptions/ApiException.java
package com.emerchantpay.gateway.genesisandroid.api.exceptions;


import com.emerchantpay.gateway.genesisandroid.api.constants.ErrorCodes;

public class ApiException extends GenesisException {

	private static final long serialVersionUID = 1L;

	String description;

	public ApiException(Integer code, String message, Throwable cause) {
		super(code, message, cause);

		description = ErrorCodes.getErrorDescription(code);
		message = description;
	}

	public ApiException() {
		super();
	}
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/exceptions/ApiException.kt
package com.emerchantpay.gateway.genesisandroid.api.exceptions


import com.emerchantpay.gateway.genesisandroid.api.constants.ErrorCodes

class ApiException : GenesisException {

    override lateinit var message: String
    private lateinit var description: String

    constructor(code: Int?, message: String, cause: Throwable) : super(code, message, cause) {
        description = ErrorCodes.getErrorDescription(code!!)
        this.message = description
    }

    constructor() : super() {}

    companion object {

        private const val serialVersionUID = 1L
    }
}
